crypto/src/conversation.rs

```diff
@@ -341,6 +345,48 @@ impl MlsConversation {
         Ok(message)
     }
 
+    fn get_self_index(&self, backend: &MlsCryptoProvider) -> CryptoResult<usize> {
+        let myself = self
+            .group
+            .key_package_ref()
+            .ok_or(CryptoError::SelfKeypackageNotFound)?;
+
+        // TODO: switch to `try_find` when stabilized
+        // let self_leaf_index = members
+        //     .into_iter()
+        //     .enumerate()
+        //     .try_find(|(index, member)| {
+        //         let member_ref = member.hash_ref(backend.crypto()).map_err(MlsError::from)?;
+        //         Ok(member_ref == *myself)
+        //     })?
+        //     .map(|(index, _)| index as u64)
+        //     .ok_or(CryptoError::SelfKeypackageNotFound)?;
+        let mut enumeration = self.group.members().into_iter().enumerate();
+        let self_leaf_index = loop {
+            if let Some((index, member)) = enumeration.next() {
+                let member_ref = member.hash_ref(backend.crypto()).map_err(MlsError::from)?;
+                if member_ref == *myself {
+                    break index;
+                }
+            } else {
+                return Err(CryptoError::SelfKeypackageNotFound);
+            }
+        };
+        Ok(self_leaf_index)
+    }
+
+    fn calculate_delay(self_index: usize, epoch: u64, total_members: usize) -> u64 {
+        let self_index = self_index as u64;
+        let total_members = total_members as u64;
+        let position = ((epoch + self_index) % total_members) + 1;
```

Comment:
TIL [modulo is distributive](https://en.wikipedia.org/wiki/Modulo_operation) i.e. `(a + b) mod n = [(a mod n) + (b mod n)] mod n`  
This would prevent your method from failing if `epoch + self_index` were to overflow